src/Components/Web.JS/src/Platform/WebAssemblyResourceLoader.ts

```diff
@@ -12,19 +12,13 @@ export class WebAssemblyResourceLoader {
       credentials: 'include',
       cache: networkFetchCacheMode
     });
+    const bootConfig: BootJsonData = await bootConfigResponse.json();
+    const cache = await getCacheToUseIfEnabled(bootConfig);
 
-    // Define a separate cache for each base href, so we're isolated from any other
-    // Blazor application running on the same origin. We need this so that we're free
-    // to purge from the cache anything we're not using and don't let it keep growing,
-    // since we don't want to be worst offenders for space usage.
-    const relativeBaseHref = document.baseURI.substring(document.location.origin.length);
-    const cacheName = `blazor-resources-${relativeBaseHref}`;
-    return new WebAssemblyResourceLoader(
-      await bootConfigResponse.json(),
-      await caches.open(cacheName));
+    return new WebAssemblyResourceLoader(bootConfig, cache);
   }
 
-  constructor (public readonly bootConfig: BootJsonData, private cache: Cache)
+  constructor (public readonly bootConfig: BootJsonData, private cacheIfUsed: Cache | null)
```

Comment:
I kinda prefer calling this `cache` and relying on the nullability.

Reply:
I've kept the scope of `cacheIfUsed` as tight as possible - it explicitly creates a separate variable called `cache` when nonnull. So I kind of like having this extra clarity here that a null value is not a random error to be worked around via saying `this.cache!`, but is an explicit scenario we need to support for good reasons.

If you feel strongly let me know, but for now I'll proceed as-is.